code-examples/python.py

```diff
@@ -1,45 +1,72 @@
-from hashlib import sha1
-from time import time
-import base64
-import urllib2
-import urllib
-import json
-import hmac
-
 """
 author: Evan Duffield <eduffield@iacquire.com>
-
 Thank you to iAcquire for sponsoring development of this module.
+
+Updated by Bryan Minor, Ph.D.
+--> Now works for Python 3
+Supported by envision.ai
 """
 
-follower_wonk_access_id = ""
-follower_wonk_secret_key = ""
+# External libraries
+from hashlib import sha1
+from time import time
+import base64
+import urllib.request
+import urllib.parse
+import urllib.error
+import json
+import hmac
 
+#Initializations
+follower_wonk_access_id_str = "member-ZWZkMDVhODItZGYwOS01YWU4LWJmMzAtOWNhM2QxZjQ3MjM1"
+follower_wonk_secret_key_str = "9wwmbjpcugzgaxuljaqnqzwte9dohjwm"
 
+#Class for access FollowerWonk API
 class FollowerWonk(object):
     @staticmethod
     def social_authority(username):
         uri = 'https://api.followerwonk.com/social-authority'
 
         datime = int(time() + 500)
 
-        s = hmac.new(follower_wonk_secret_key, "%s\n%s" % (follower_wonk_access_id, datime), sha1).digest()
+
+        #Add BMM: Trying to binary format correct
+        keyBin = bytes(follower_wonk_secret_key_str, 'UTF-8')
+        messageStr = "%s\n%s" % (follower_wonk_access_id_str, datime)
+        messageBin = bytes(messageStr, 'UTF-8')
+
+
+        s = hmac.new(keyBin, messageBin, sha1).digest()
         b64 = base64.b64encode(s)
-        signature = urllib.quote_plus(b64)
+        signature = urllib.parse.quote_plus(b64)
+
+        auth = "AccessID=%s;Timestamp=%s;Signature=%s;" % (follower_wonk_access_id_str, datime, signature)
 
-        auth = "AccessID=%s;Timestamp=%s;Signature=%s;" % (follower_wonk_access_id, datime, signature)
+        req = urllib.request.Request("%s?screen_name=%s;%s" % (uri, username, auth))
+        r = urllib.request.urlopen(req)
 
-        req = urllib2.Request("%s?screen_name=%s;%s" % (uri, username, auth))
-        r = urllib2.urlopen(req)
+        responseStr = r.read().decode("utf-8")
+        #print("responseStr = ", responseStr)
+        response_Json = json.loads(responseStr)
 
-        response = json.loads(r.read())
+        #print("response_Json = ", response_Json)
 
         r.close()
 
-        if '_embedded' not in response:
+        if '_embedded' not in response_Json:
             return -1
 
-        return response['_embedded'][0]['social_authority']
+        return float(response_Json['_embedded'][0]['social_authority'])
+        #return response_Json
+
+assert FollowerWonk.social_authority("bryanminorphd") > 0
```

Comment:
It's strange to include these in a module. In theory, someone could `import` this code and the `import` would execute these assertions which is not desired behavior.